addons/mail/static/tests/mail_utils_tests.js

```diff
@@ -35,6 +35,28 @@ QUnit.test('add_link utility function', function (assert) {
     });
 });
 
+QUnit.test('addLink: utility function and special entities', function (assert) {
+    assert.expect(4);
+
+    var testInputs = {
+        // textContent not unescaped
+        '<p>https://example.com/?&amp;currency_id</p>':
+        '<p><a target="_blank" rel="noreferrer noopener" href="https://example.com/?&amp;currency_id">https://example.com/?&amp;currency_id</a></p>',
+        // entities not unescaped
+        '&amp; &amp;amp; &gt; &lt;': '&amp; &amp;amp; &gt; &lt;',
+        // &gt; not linkified since it is not in URL regex
+        '<p>https://example.com/&gt;</p>':
+        '<p><a target="_blank" rel="noreferrer noopener" href="https://example.com/">https://example.com/</a>&gt;</p>',
+        // special character in smileys should be escaped
+        ':\'(': ':&#x27;(',
```

Comment:
I don't agree with this part, why should "add link" function escape anything?
It should keep escaped what is already escaped, but it should not escape extra things.

I think the problem comes from `_.escape()` escaping `'` and `"` in context where it is inoffensive. HTML escaping should be done differently depending on if the target is an attribute or a node content.

The subsequent injection done in `addLink` is unable to "unescape" the wrongly escaped `"` and `'`.